__tests__/app.test.js

```diff
@@ -15,158 +15,198 @@ afterAll( () => {
+              expect.objectContaining({
+                comment_id: expect.any(Number),
+                votes: expect.any(Number),
+                created_at: expect.any(String),
+                author: expect.any(String),
+                body: expect.any(String),
+                article_id: article_id, 
+              })
+            );
+          });
         });
-      });
-  });
-  test('200: Responds with an array of comments for article_id', () => {
-    const article_id = 0;
-    return request(app)
-      .get(`/api/articles/${article_id}/comments`)
-      .expect(200)
-      .then(({ body: { comments } }) => {
-        expect(Array.isArray(comments)).toBe(true); 
-        expect(comments).toHaveLength(0); 
-      });
-  });
-  test('200: Responds with the comments in the correct order (last is first)', () => {
-    const article_id = 1; 
-    return request(app)
-      .get(`/api/articles/${article_id}/comments`)
-      .expect(200)
-      .then(({ body: { comments } }) => {
-        expect(comments[0].comment_id).toBeGreaterThan(comments[1].comment_id); 
-      });
+    });
+    test('200: Responds with an array of comments for article_id', () => {
+      const article_id = 0;
+      return request(app)
+        .get(`/api/articles/${article_id}/comments`)
+        .expect(200)
+        .then(({ body: { comments } }) => {
+          expect(Array.isArray(comments)).toBe(true); 
+          expect(comments).toHaveLength(0); 
+        });
+    });
+    test('200: Responds with the comments in the correct order (last is first)', () => {
+      const article_id = 1; 
+      return request(app)
+        .get(`/api/articles/${article_id}/comments`)
+        .expect(200)
+        .then(({ body: { comments } }) => {
+          expect(comments[0].comment_id).toBeGreaterThan(comments[1].comment_id); 
+        });
+    });
   });
 });
+
+describe('POST - Endpoints', () => {
+  describe('POST /api/articles/:article_id/comments', () => {
+    test('201: Responds with the posted comment', () => {
+        return request(app)
+            .post('/api/articles/1/comments')
+            .send({ username: 'butter_bridge', body: 'Great article!' })
+            .expect(201)
+            .then(({ body: { comment } }) => {
+                expect(comment).toMatchObject({
+                    comment_id: expect.any(Number),
+                    body: 'Great article!',
+                    article_id: 1,
+                    author: 'butter_bridge',
+                    votes: 0,
+                    created_at: expect.any(String),
+                });
+            });
+    });
+    test('400: Responds with "Bad request" when missing username or body', () => {
+      return request(app)
+          .post('/api/articles/1/comments')
+          .send({ username: 'some_name' })
+          .expect(400)
+          .then(({ body: { msg } }) => {
+              expect(msg).toBe('Bad request: missing username or body');
+          });
+    });
```

Comment:
You're missing several tests here. What about posting to an article that doesn't exist, an article with a bad id, a missing comment, a comment with no body. You will need to revisit this.